acceptance/tests/api-gateway/api_gateway_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package apigateway
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/consul-k8s/acceptance/framework/consul"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/k8s"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	StaticClientName           = "static-client"
+	gatewayClassControllerName = "hashicorp.com/consul-api-gateway-controller"
+	gatewayClassFinalizer      = "gateway-exists-finalizer.consul.hashicorp.com"
+	gatewayFinalizer           = "gateway-finalizer.consul.hashicorp.com"
+)
+
+// Test that api gateway basic functionality works in a default installation and a secure installation.
+func TestAPIGateway(t *testing.T) {
+	cases := []struct {
+		secure bool
+	}{
+		{
+			secure: false,
+		},
+		{
+			secure: true,
+		},
+	}
+	for _, c := range cases {
+		name := fmt.Sprintf("secure: %t", c.secure)
+		t.Run(name, func(t *testing.T) {
+			ctx := suite.Environment().DefaultContext(t)
+			cfg := suite.Config()
+			helmValues := map[string]string{
+				"connectInject.enabled":        "true",
+				"global.acls.manageSystemACLs": strconv.FormatBool(c.secure),
+				"global.tls.enabled":           strconv.FormatBool(c.secure),
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			logger.Log(t, "creating api-gateway resources")
+			out, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "apply", "-k", "../fixtures/bases/api-gateway")
+			require.NoError(t, err, out)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
+				// Ignore errors here because if the test ran as expected
+				// the custom resources will have been deleted.
+				k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "-k", "../fixtures/bases/api-gateway")
+			})
+
+			logger.Log(t, "creating target server")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/cases/static-server-inject")
+
+			// TODO: patch routes to point to the target
+
+			// We use the static-client pod so that we can make calls to the api gateway
+			// via kubectl exec without needing a route into the cluster from the test machine.
+			logger.Log(t, "creating static-client pod")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.DebugDirectory, "../fixtures/bases/static-client")
+
+			// With the cluster up, we can use the client to verify behavior in Consul
+			consulClient, _ := consulCluster.SetupConsulClient(t, c.secure)
+			_ = consulClient
```

Comment:
just leaving this here for now since eventually we'll want to validate the objects that get created in Consul as well.